mongodbatlas/fw_provider.go

```diff
@@ -0,0 +1,394 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/int64validator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/listvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/provider"
+	"github.com/hashicorp/terraform-plugin-framework/provider/schema"
+	"github.com/hashicorp/terraform-plugin-framework/providerserver"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-mux/tf5to6server"
+	"github.com/hashicorp/terraform-plugin-mux/tf6muxserver"
+	sdkv2schema "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	cstmvalidator "github.com/mongodb/terraform-provider-mongodbatlas/mongodbatlas/framework/validator"
+)
+
+type MongodbtlasProvider struct{}
+
+type tfMongodbAtlasProviderModel struct { //nolint:govet //prefer to maintain order of attributes over small byte optimization
+	PublicKey            types.String `tfsdk:"public_key"`
+	PrivateKey           types.String `tfsdk:"private_key"`
+	BaseURL              types.String `tfsdk:"base_url"`
+	RealmBaseURL         types.String `tfsdk:"realm_base_url"`
+	IsMongodbGovCloud    types.Bool   `tfsdk:"is_mongodbgov_cloud"`
+	AssumeRole           types.List   `tfsdk:"assume_role"`
+	SecretName           types.String `tfsdk:"secret_name"`
+	Region               types.String `tfsdk:"region"`
+	StsEndpoint          types.String `tfsdk:"sts_endpoint"`
+	AwsAccessKeyID       types.String `tfsdk:"aws_access_key_id"`
+	AwsSecretAccessKeyID types.String `tfsdk:"aws_secret_access_key"`
+	AwsSessionToken      types.String `tfsdk:"aws_session_token"`
+}
+
+type tfAssumeRoleModel struct { //nolint:govet //prefer to maintain order of attributes over small byte optimization
```

Comment:
I don't like the `tf` |`fw`. Can we use framework instead?

Reply:
I did like the tf convention as it is concise and quickly lets you know that the model is used for handling data to/from terraform. We can carry the conversation over here https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1365#discussion_r1284391764 as I believe it is the same.